### PR TITLE
WFLY-333 Deploying a jdbc driver ignores the --name parameter in the cli

### PR DESCRIPTION
**Pull request by maeste**

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
WFLY-333 does not really seem relevant, I think it should be WFLY-3333.

File: connector/src/main/java/org/jboss/as/connector/deployers/ds/processors/DriverProcessor.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                                 Integer.valueOf(minorVersion));
                     }
                     String driverName = deploymentUnit.getName();
-                    if (! deploymentUnit.getName().endsWith(".jar") || driverNames.size() != 1) {
+                    if ((deploymentUnit.getName().contains(".") && ! deploymentUnit.getName().endsWith(".jar")) || driverNames.size() != 1) {
```
Comment (stuartwdouglas) [inline connector/src/main/java/org/jboss/as/connector/deployers/ds/processors/DriverProcessor.java]: I'm not really sure if this is a good idea. It seems very odd that this behaviour is determined by the presence of a '.' in the deployment name, although I can't really think of any other way for the user to specify the driver name.

Comment (bstansberry) [inline connector/src/main/java/org/jboss/as/connector/deployers/ds/processors/DriverProcessor.java]: I can't think of any other way either. It's odd, but I think it's ok so long as it's documented and will be supported.

We use similar naming convention stuff in lots of other places. E.g. ending in ".war" triggers all sort of war-related processing, ".ear" triggers ear-related processing.

**Comment by bstansberry** [comment]
No one has any other ideas in a month, so this is it. :)

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**